issues.js

```diff
@@ -168,23 +189,32 @@ function groupIssuesByTopic(issues) {
 // Append the give issue to the appropriate Topic list
 function renderIssue(issue) {
     var rendered_issue = Handlebars.templates.issues(issue);
-    var $topic_group = $(sanitizeId(issue.topic));
+    var $topic_group = $(sanitizeId(issue.category) + " " + sanitizeId(issue.topic));
     $topic_group.removeClass("hide");
     $topic_group.children("ul").append(rendered_issue);
 }
 
 // Render the containers for each Topic list
-function renderTopics(topics) {
-    var topic_obj = {};
-    topic_obj.topics = topics;
-    
-    var rendered_topics = Handlebars.templates.topic_groups(topic_obj);
-    $("#issues_list").html(rendered_topics);
+function renderGroupings(topics, categories) {
+    var rendered_topics = Handlebars.templates.topic_groups({
+      topics: topics
+    }); // render a topics snippet to be used by each category
+
+    var rendered_categories = Handlebars.templates.categories({
+      categories: categories
+    }); // render the categories
+
+    $("#null_category").html(rendered_topics); // a bucket for all issues without a category
+
+    $("#issues_list").append(rendered_categories);
+    $.each(categories, function (key, category) {
+      $(sanitizeId(category)).append(rendered_topics);
```

Comment:
It looks like this is rendering all the topics into each category and hiding the ones that shouldn't be shown?

I see duplicate elements and IDs so the logic needs tweaking here:

![duckduckhack_tasks](https://cloud.githubusercontent.com/assets/873785/23634480/5b9c2244-0298-11e7-83ab-58896284f86e.png)

Perhaps we should build an array of objects containing categories and their issues. That might also simplify things so we only need a single template?

E.g something like...

```js
var categories = [{
  name: "top",
  topics: [
    { name: "foo", id: "top_foo" issues: [ { issue1: ... }, { issue2: ... } ]
    { name: "bar", id: "top_bar" issues: [ { issue1: ... }, { issue2: ... } ]
  ]
}, {
  name: "Programming Mission",
  topics: [
    { name: foo2, issues: [ { issue1: ... }, { issue2: ... } ]
    { name: bar2, issues: [ { issue1: ... }, { issue2: ... } ]
  ]
}];
```

One other thing is that I think we need more logic for handling the `ID` generation. I think we should prefix the category to the topic name in case multiple categories have the same ID.